### ISSUE TITLE
Tuple-expansion with nil to vararg lib function crashes compiler

### ISSUE THREAD

**Issue by Papierkorb** [crystal-lang/crystal#4666]
Hello,

When expanding a tuple containing a `nil` into a `lib` vararg function, the compiler crashes with an invalid memory access.

The following code crashes the compiler (See below):

```crystal
lib Foo
  # Some vararg function
  fun bar(baz : Int32, ...) : Int32
end

def fwd(*arguments)
  Foo.bar(*arguments) # Expand argument tuple with nil
end

fwd(1, nil) # Crashes
# Foo.bar(*{1, nil}) # Crashes
# Foo.bar(1, nil) # Works fine
# fwd(1, 2) # Works fine
```

## Compiler log

```
Invalid memory access (signal 11) at address 0x0
[0x43cebf] __crystal_sigfault_handler +63
[0x7fe9578c6940] ???
[0x7fe958ebeee4] _ZN4llvm11PointerType3getEPNS_4TypeEj +20
[0x7fe958dea911] _ZN4llvm25ConstantFoldGetElementPtrEPNS_4TypeEPNS_8ConstantEbNS_8OptionalIjEENS_8ArrayRefIPNS_5ValueEEE +1937
[0x7fe958e07471] _ZN4llvm12ConstantExpr16getGetElementPtrEPNS_4TypeEPNS_8ConstantENS_8ArrayRefIPNS_5ValueEEEbNS_8OptionalIjEES2_ +113
[0x7fe958e1333b] LLVMBuildInBoundsGEP +619
[0x133c426] ???
[0x13a7a63] ???
[0x139a91c] ???
[0x1364a7e] ???
[0x1364e26] ???
[0x1386c93] ???
[0x138d294] ???
[0x139c799] ???
[0x1364a7e] ???
[0x1364e26] ???
[0x12fc8ad] ???
[0x1b92f85] ???
[0x1c372e9] ???
[0x475dfd] ???
[0x43c7ed] main +18909
[0x7fe956cc343a] __libc_start_main +234
[0x4097da] _start +42
[0x0] ???
```

**Platform**: `Crystal 0.22.0 (2017-04-22) LLVM 4.0.0` on `ArchLinux x64`

Thanks.

**Comment by sdogruyol** [comment]
On 0.23.1, LLVM 4.0 Arch x 64 this just errors.

```
_main.o: In function `fwd':
/home/sdogruyol/playground/crystal/bug.cr:7: undefined reference to `bar'
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o '/home/sdogruyol/.cache/crystal/crystal-run-bug.tmp'  -rdynamic  -lpcre -lgc -lpthread /home/sdogruyol/playground/crystal/crystal/src/ext/libcrystal.a -levent -lrt -ldl -L/usr/lib -L/usr/local/lib`
```

**Comment by Papierkorb** [comment]
@sdogruyol I can't reproduce this anymore either.  I'll assume this was "accidentally" fixed by some other change since version `0.22`.
